// chalk-ex@1.0.1/lib/index.js
import chalk from 'chalk';

var color = {
  red: 'crimson',
  pink: 'violet',
  purple: 'darkviolet',
  deepPurple: 'blueviolet',
  indigo: 'royalblue',
  blue: 'dodgerblue',
  lightBlue: 'deepskyblue',
  cyan: 'darkturquoise',
  teal: 'mediumseagreen',
  green: 'lime',
  lightGreen: 'greenyellow',
  lime: 'khaki',
  yellow: 'yellow',
  amber: 'gold',
  orange: 'orange',
  deepOrange: 'coral',
  brown: 'rosybrown',
  gray: 'darkgray',
  blueGray: 'lightslategray',
  white: 'white',
  black: 'black'
};
var label$1 = {
  log: color.white,
  info: color.blue,
  success: color.green,
  warn: color.orange,
  error: color.red
};
var config = {
  color: color,
  label: label$1
};

function createLabel(content, color) {
  return chalk.black.bgKeyword(color)(" ".concat(content, " "));
}

function createColorLabel(color) {
  return function (content) {
    return createLabel(content, color);
  };
}

function createColorText(color) {
  return function (content) {
    return chalk.keyword(color)(content);
  };
}

var label = {};
var text = {};

for (var l in config.label) {
  label[l] = createLabel(l.toUpperCase(), config.label[l]);
}

for (var c in config.color) {
  label[c] = createColorLabel(config.color[c]);
  text[c] = createColorText(config.color[c]);
}

export { label, text };
